tests/test_pages_price.py

```diff
@@ -0,0 +1,244 @@
+import pytest
+from web_poet import HttpResponse, field
+
+from zyte_common_items import ProductPage
+
+
+@pytest.mark.asyncio
+async def test_price_selector():
+    class CustomProductPage(ProductPage):
+        call_count = 0
+
+        @field
+        def price(self):
+            self.call_count += 1
+            return self.css("div::text")
+
+    html = b"""
+    <!DOCTYPE html>
+    <html>
+        <body>
+            <div>$13.2</div>
+        </body>
+    </html>
+    """
+    url = "https://example.com"
+    page = CustomProductPage(response=HttpResponse(url=url, body=html))
+    assert page.call_count == 0
+    assert page.price == "13.2"
+    assert page.call_count == 1
+    assert page.currency is None
+    assert await page.currencyRaw == "$"
+    assert page.call_count == 1
+
+    # access currency fields before the price field
+    page = CustomProductPage(response=HttpResponse(url=url, body=html))
+    assert page.call_count == 0
+    assert page.currency is None
+    assert await page.currencyRaw == "$"
+    assert page.call_count == 1
+    assert page.price == "13.2"
+    assert page.call_count == 2
+
+
+@pytest.mark.asyncio
+async def test_price_explicit():
+    class CustomProductPage(ProductPage):
+        @field
+        def price(self):
+            return self.css("div::text").get()
+
+    html = b"""
+    <!DOCTYPE html>
+    <html>
+        <body>
+            <div>$13.2</div>
+        </body>
+    </html>
+    """
+    url = "https://example.com"
+    page = CustomProductPage(response=HttpResponse(url=url, body=html))
+    assert page.price == "$13.2"
+    assert page.currency is None
+    assert await page.currencyRaw is None
```

Comment:
Would it make sense to check the call count in this test as well? The code path looks different enough.